torch/csrc/jit/passes/quantization.cpp

```diff
@@ -23,7 +72,30 @@ void InsertObserverNodes(std::shared_ptr<Graph>& graph) {
 }
 
 void InsertFakeQuantNodes(std::shared_ptr<Graph>& graph) {
-  throw std::runtime_error("Pass not implemented yet!");
+  for (const auto& n : graph->nodes()) {
+    // Skip nodes that we've just added
+    if ((n->kind().toQualString() == std::string("aten::quant")) ||
+      (n->kind().toQualString() == std::string("aten::dequant"))) {
+      continue;
+    }
+    if (checkIfNodeValuesQuantizable(n) != 0) {
+      for (torch::jit::Value* v : n->outputs()) {
+        if (v->isCompleteTensor()) {
+          addQuantDeQuantNodes(v);
+        }
+      }
+      // This avoid duplicate insertion of quant-dequant pair for consecutive
+      // quantizable nodes which do not form pattern
+      if (n->prev() != nullptr && n->prev()->kind().toQualString() !=
```

Comment:
There is no guarantees `n->prev` would be a dequant node even if the value was quantized - we should check for `v->def` instead.